src/profile-logic/profile-data.js

```diff
@@ -1312,8 +1338,18 @@ export function getSampleIndexClosestToTime(
 
   // Check the distance between the provided time and the center of the bisected sample
   // and its predecessor.
-  const distanceToThis = samples.time[index] + interval / 2 - time;
-  const distanceToLast = time - (samples.time[index - 1] + interval / 2);
+  const previousIndex = index - 1;
+
+  let duration = interval;
+  let previousDuration = interval;
+  if (samples.duration) {
+    duration = samples.duration[index];
+    previousDuration = samples.duration[previousIndex];
+  }
```

Comment:
I'm not 100% sure I shouldn't use the interval always here. It doesn't seem right to measure a distance using a negative duration...

Reply:
Could you take the absolute value of the duration? This would handle both the diffing case and the JS tracer case.

Reply:
yeah this sounds right